src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -21,93 +24,85 @@ public QueryTimeouts(IDocumentStore documentStore, string endpointName)
             TriggerCleanupEvery = TimeSpan.FromMinutes(2);
             CleanupGapFromTimeslice = TimeSpan.FromMinutes(1);
             shutdownTokenSource = new CancellationTokenSource();
+            logger = LogManager.GetLogger<QueryTimeouts>();
         }
 
         public TimeSpan CleanupGapFromTimeslice { get; set; }
         public TimeSpan TriggerCleanupEvery { get; set; }
+        public Func<DateTime> GetUtcNow { get; set; } = () => DateTime.UtcNow;
 
         public async Task<TimeoutsChunk> GetNextChunk(DateTime startSlice)
         {
-            var now = DateTime.UtcNow;
+            var now = GetUtcNow();
             List<TimeoutsChunk.Timeout> results;
 
             // Allow for occasionally cleaning up old timeouts for edge cases where timeouts have been
             // added after startSlice have been set to a later timout and we might have missed them
             // because of stale indexes.
             if (lastCleanupTime == DateTime.MinValue || lastCleanupTime.Add(TriggerCleanupEvery) < now)
             {
-                results = await GetCleanupChunk(startSlice).ConfigureAwait(false);
+                results = await GetCleanupChunk(now).ConfigureAwait(false);
             }
             else
             {
                 results = new List<TimeoutsChunk.Timeout>();
             }
 
             // default return value for when no results are found
-            var nextTimeToRunQuery = DateTime.UtcNow.AddMinutes(10);
-
-            RavenQueryStatistics statistics;
+            var nextTimeoutToExpire = now.AddMinutes(10);
 
             using (var session = documentStore.OpenAsyncSession())
             {
-                int totalCount, skipCount = 0;
+                // This is all an unexecuted Raven query expression
```

Comment:
@weralabaj: Not just for you! A non-executed Raven query expression is a good reminder to have no matter how familiar you are with Raven. Keeps people from trying to run in and optimize away LINQ or do other things which you can't do.